entity-service-impl/src/main/java/org/hypertrace/entity/service/util/DocStoreConverter.java

```diff
@@ -128,17 +128,13 @@ public static Filter getTenantIdEqFilter(String tenantId) {
 
   private static Filter transform(AttributeFilter filter) {
     if (filter.hasAttributeValue()) {
-      if (ATTRIBUTES_LABELS_FIELD_NAME.equals(filter.getName())
-          && filter.getOperator() == Operator.EQ) {
+      if (isMultiValuedAttribute(filter) && filter.getOperator() == Operator.EQ) {
```

Comment:
OK, got the issue - because EQS goes through EDS (which has been one of our longest standing tech debts) the metadata gets lost.

So our options, are 
1. Fix that now and have EQS go directly to doc store
2. Find some way of reading metadata in EDS, but difficult since we're working in sub doc paths at this point
3. Pass this info along, for example via filter as implemented here.

I think everyone can guess my preference 😉 , but I get why it's easier to do 3 and I'm fine with it - but I do have a question - does anyone know was this tech debt was initially incurred? The translation from EQS->EDS looks just as involved as the translation from EDS->doc store, so what did we save by doing it?

Reply:
I plead guilty for adding that tech debt, but honestly can't remember why I did that.

Reply:
Certainly been in the same boat! It may be worth a 5m eval to just see if it would be hard to migrate now - IMO it's worth the extra work to avoid changing the EDS protos if we're talking hours and not days.

Reply:
With less context on Entity and Attribute Service and there is strict deadline for this feature from customer, don't want to do this at this point of time. Can someone of you please take the call ?

Reply:
This change would be internal to ES (which is why I like it vs the proto change which is external) and not involve AS - it would just be changing the conversion from EQS->EDS to an EQS->doc store and calling the doc store directly. I'm fine with either call - if you think it will significantly impact the time to deliver the feature, we don't have to address here.